<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>R-side access to published microbial signatures from BugSigDB â€¢ bugsigdbr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="R-side access to published microbial signatures from BugSigDB">
<meta property="og:description" content="bugsigdbr">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bugsigdbr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/bugsigdbr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/bugsigdbr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="bugsigdbr_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>R-side access to published microbial signatures from BugSigDB</h1>
                        <h4 data-toc-skip class="author">Ludwig Geistlinger, Jennifer Wokaty, and Levi Waldron</h4>
            
            <h4 data-toc-skip class="date">24 June 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/bugsigdbr/blob/master/vignettes/bugsigdbr.Rmd" class="external-link"><code>vignettes/bugsigdbr.Rmd</code></a></small>
      <div class="hidden name"><code>bugsigdbr.Rmd</code></div>

    </div>

    
    
<div id="bugsigdb-a-comprehensive-database-of-published-microbial-signatures" class="section level1">
<h1 class="hasAnchor">
<a href="#bugsigdb-a-comprehensive-database-of-published-microbial-signatures" class="anchor" aria-hidden="true"></a>BugSigDB: a comprehensive database of published microbial signatures</h1>
<p><a href="https://bugsigdb.org" class="external-link">BugSigDB</a> is a manually curated database of microbial signatures from the published literature of differential abundance studies of human and other host microbiomes.</p>
<p>BugSigDB provides:</p>
<ul>
<li>standardized data on geography, health outcomes, host body sites, and experimental, epidemiological, and statistical methods using controlled vocabulary,</li>
<li>results on microbial diversity,</li>
<li>microbial signatures standardized to the NCBI taxonomy, and</li>
<li>identification of published signatures where a microbe has been reported.</li>
</ul>
<p>The <a href="https://github.com/waldronlab/bugsigdbr" class="external-link">bugsigdbr</a> package implements convenient access to BugSigDB from within R/Bioconductor. The goal of the package is to facilitate import of BugSigDB data into R/Bioconductor, provide utilities for extracting microbe signatures, and enable export of the extracted signatures to plain text files in standard file formats such as <a href="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29" class="external-link">GMT</a>.</p>
<p>The <a href="https://github.com/waldronlab/bugsigdbr" class="external-link">bugsigdbr</a> package is primarily a data package. For descriptive statistics and comprehensive analysis of BugSigDB contents, please see the <a href="https://github.com/waldronlab/BugSigDBStats" class="external-link">BugSigDBStats package</a> and <a href="http://waldronlab.io/BugSigDBStats/articles/BugSigDBStats.html" class="external-link">analysis vignette</a>.</p>
<p>We start by loading the package</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/bugsigdbr" class="external-link">bugsigdbr</a></span><span class="op">)</span></code></pre></div>
<div id="obtaining-published-microbial-signatures-from-bugsigdb" class="section level2">
<h2 class="hasAnchor">
<a href="#obtaining-published-microbial-signatures-from-bugsigdb" class="anchor" aria-hidden="true"></a>Obtaining published microbial signatures from BugSigDB</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bsdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/importBugSigDB.html">importBugSigDB</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2021-06-24 20:07:55</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">bsdb</span><span class="op">)</span>
<span class="co">#&gt;  [1] "Study"                      "Study design"              </span>
<span class="co">#&gt;  [3] "PMID"                       "DOI"                       </span>
<span class="co">#&gt;  [5] "URL"                        "Authors"                   </span>
<span class="co">#&gt;  [7] "Title"                      "Journal"                   </span>
<span class="co">#&gt;  [9] "Year"                       "Abstract"                  </span>
<span class="co">#&gt; [11] "Experiment"                 "Location of subjects"      </span>
<span class="co">#&gt; [13] "Host species"               "Body site"                 </span>
<span class="co">#&gt; [15] "Condition"                  "Group 0 name"              </span>
<span class="co">#&gt; [17] "Group 1 name"               "Group 1 definition"        </span>
<span class="co">#&gt; [19] "Group 0 sample size"        "Group 1 sample size"       </span>
<span class="co">#&gt; [21] "Antibiotics exclusion"      "Sequencing type"           </span>
<span class="co">#&gt; [23] "16s variable region"        "Sequencing platform"       </span>
<span class="co">#&gt; [25] "Statistical test"           "Significance threshold"    </span>
<span class="co">#&gt; [27] "MHT correction"             "LDA Score above"           </span>
<span class="co">#&gt; [29] "Matched on"                 "Confounders controlled for"</span>
<span class="co">#&gt; [31] "Pielou"                     "Shannon"                   </span>
<span class="co">#&gt; [33] "Chao1"                      "Simpson"                   </span>
<span class="co">#&gt; [35] "Inverse Simpson"            "Richness"                  </span>
<span class="co">#&gt; [37] "Signature page name"        "Source"                    </span>
<span class="co">#&gt; [39] "Curated date"               "Curator"                   </span>
<span class="co">#&gt; [41] "Revision editor"            "Description"               </span>
<span class="co">#&gt; [43] "Abundance in Group 1"       "MetaPhlAn taxon names"     </span>
<span class="co">#&gt; [45] "NCBI Taxonomy IDs"</span></code></pre></div>
</div>
<div id="extracting-microbe-signatures" class="section level2">
<h2 class="hasAnchor">
<a href="#extracting-microbe-signatures" class="anchor" aria-hidden="true"></a>Extracting microbe signatures</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sigs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getSignatures.html">getSignatures</a></span><span class="op">(</span><span class="va">bsdb</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">sigs</span><span class="op">)</span>
<span class="co">#&gt; [1] 1802</span>
<span class="va">sigs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>
<span class="co">#&gt; $`bsdb:1/1/1_adenoma:conventional-adenoma-cases_vs_controls_UP`</span>
<span class="co">#&gt;  [1] "91061"  "1236"   "1654"   "1716"   "1301"   "162289" "189330" "33024" </span>
<span class="co">#&gt;  [9] "40544"  "2037"   "2049"   "506"    "186826" "1300"   "31977"  "91347" </span>
<span class="co">#&gt; [17] "1653"   "57037"  "1386"   "186817"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`bsdb:1/1/2_adenoma:conventional-adenoma-cases_vs_controls_DOWN`</span>
<span class="co">#&gt; [1] "100883" "1117"  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`bsdb:1/2/1_hyperplastic-polyp:hyperplastic-polyp-cases_vs_controls_UP`</span>
<span class="co">#&gt; [1] "207244" "57037"</span></code></pre></div>
</div>
<div id="writing-microbe-signatures-to-file-in-gmt-format" class="section level2">
<h2 class="hasAnchor">
<a href="#writing-microbe-signatures-to-file-in-gmt-format" class="anchor" aria-hidden="true"></a>Writing microbe signatures to file in GMT format</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/writeGMT.html">writeGMT</a></span><span class="op">(</span><span class="va">sigs</span>, gmt.file <span class="op">=</span> <span class="st">"bugsigdb_signatures.gmt"</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; [1] TRUE</code></pre>
</div>
<div id="displaying-bugsigdb-signature-and-taxon-pages" class="section level2">
<h2 class="hasAnchor">
<a href="#displaying-bugsigdb-signature-and-taxon-pages" class="anchor" aria-hidden="true"></a>Displaying BugSigDB signature and taxon pages</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/browseSignature.html">browseSignature</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sigs</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; [1] "https://bugsigdb.org/Study_1/Experiment_1/Signature_1"</span>
<span class="fu"><a href="../reference/browseTaxon.html">browseTaxon</a></span><span class="op">(</span><span class="va">sigs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; [1] "https://bugsigdb.org/Special:RunQuery/Taxon?Taxon%5BNCBI%5D=91061&amp;_run=1"</span></code></pre></div>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor" aria-hidden="true"></a>Session info</h1>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.0 (2021-05-18)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.2 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] bugsigdbr_0.99.0 BiocStyle_2.21.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Rcpp_1.0.6          pillar_1.6.1        bslib_0.2.5.1      </span>
<span class="co">#&gt;  [4] compiler_4.1.0      BiocManager_1.30.16 jquerylib_0.1.4    </span>
<span class="co">#&gt;  [7] dbplyr_2.1.1        tools_4.1.0         digest_0.6.27      </span>
<span class="co">#&gt; [10] bit_4.0.4           tibble_3.1.2        lifecycle_1.0.0    </span>
<span class="co">#&gt; [13] jsonlite_1.7.2      evaluate_0.14       memoise_2.0.0      </span>
<span class="co">#&gt; [16] BiocFileCache_2.1.0 RSQLite_2.2.7       pkgconfig_2.0.3    </span>
<span class="co">#&gt; [19] rlang_0.4.11        DBI_1.1.1           filelock_1.0.2     </span>
<span class="co">#&gt; [22] curl_4.3.2          yaml_2.2.1          pkgdown_1.6.1.9001 </span>
<span class="co">#&gt; [25] xfun_0.24           fastmap_1.1.0       withr_2.4.2        </span>
<span class="co">#&gt; [28] httr_1.4.2          stringr_1.4.0       dplyr_1.0.7        </span>
<span class="co">#&gt; [31] knitr_1.33          rappdirs_0.3.3      desc_1.3.0         </span>
<span class="co">#&gt; [34] generics_0.1.0      fs_1.5.0            sass_0.4.0         </span>
<span class="co">#&gt; [37] vctrs_0.3.8         systemfonts_1.0.2   tidyselect_1.1.1   </span>
<span class="co">#&gt; [40] rprojroot_2.0.2     bit64_4.0.5         glue_1.4.2         </span>
<span class="co">#&gt; [43] R6_2.5.0            textshaping_0.3.5   fansi_0.5.0        </span>
<span class="co">#&gt; [46] rmarkdown_2.9       bookdown_0.22       purrr_0.3.4        </span>
<span class="co">#&gt; [49] blob_1.2.1          magrittr_2.0.1      ellipsis_0.3.2     </span>
<span class="co">#&gt; [52] htmltools_0.5.1.1   assertthat_0.2.1    ragg_1.1.3         </span>
<span class="co">#&gt; [55] utf8_1.2.1          stringi_1.6.2       cachem_1.0.5       </span>
<span class="co">#&gt; [58] crayon_1.4.1</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Ludwig Geistlinger, Levi Waldron, Jennifer Wokaty.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link external-link">pkgdown</a> 1.6.1.9001.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
